mongo/readpref/mode_test.go

```diff
@@ -0,0 +1,22 @@
+// Copyright (C) MongoDB, Inc. 2020-present.
+
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+package readpref
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestMode_String(t *testing.T) {
+	assert.Equal(t, "primary", PrimaryMode.String())
```

Comment:
We use `go.mongodb.org/mongo-driver/internal/testutil/assert` for the test assertions in our codebase. The API is similar to `testify/assert`, but it avoids the large `testify` dependency and requires error messages so the ordering of the parameters doesn't matter. There are pieces of the codebase that use `testify`, but we're working on getting rid of them.

Also, because the assertion library requires error messages, making these tests table-driven might make them less verbose.

Reply:
Done.